runtime/lua/vim/lsp/completion.lua

```diff
@@ -359,7 +359,8 @@ local function request(clients, bufnr, win, callback)
   local request_ids = {} --- @type table<integer, integer>
   local remaining_requests = vim.tbl_count(clients)
 
-  for client_id, client in pairs(clients) do
+  for _, client in pairs(clients) do
```

Comment:
To prevent confusion in the future, could you please also change the parameter type of `clients` to `vim.lsp.Client[]`? That way it's clear that we're not expecting a dictionary or doing anything with the keys.

Reply:
Thank you for your quick response and explanation! Certainly I use omnifunc.

I changed it.

Reply:
> could you please also change the parameter type of clients to vim.lsp.Client[]? That way it's clear that we're not expecting a dictionary or doing anything with the keys.

The downside of that is that it would give the impression that it's safe to use ipairs which isn't the case if the method is called from the autotrigger path.

I think adding a comment that keys!=client id would be clearer. 

Changing the type to `table<integer, vim.lsp.Client>|vim.lsp.Client[]` would be another option but somebody might take that as a mistake and try to clean it up

Reply:
I think it should be `table<unknown, vim.lsp.Client>`

Reply:
I prefer `vim.lsp.Client[]` or `table<integer, vim.lsp.Client>|vim.lsp.Client[]`. 

> I think adding a comment that keys!=client id would be clearer.

I think just only adding a comment, the role of the keys in the table is unclear and might confuse someone. (it might be because of I am beginner of neovim code base)

Reply:
I don't feel too strongly about this, let's not overthink it. Changing the type and/or adding a comment is okay, the main goal is to prevent this from happening again and clarifying the use and expected structure of the parameter.

Reply:
Okay. I only added comment.

@mfussenegger 
Comment position is okay? Or top of for-loop?